src/app/mod.rs

```diff
@@ -1227,6 +1230,32 @@ impl<'a, 'b> App<'a, 'b> {
         self.get_matches_from_safe(&mut env::args_os())
     }
 
+    /// Similar to [`App::get_matches`] but also reads args from
+    /// env-var `env_var_name`.
+    ///
+    /// Defaults to read from "VAR" if `env_var_name` is `None`.
+    pub fn get_matches_with_env(self, env_var_name: Option<String>)
```

Comment:
Shouldn't that be `<S:AsRef<str>(self, env_var_name: &S)`?
`env_var_name` isn't stored anywhere, so why the lifetime `'b`?
And `&str` doesn't implement `From<&String>`.